.github/workflows/samples-tools/pkg/utils/config.go

```diff
@@ -0,0 +1,95 @@
+package utils
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+type Config struct {
+	// Pattern to match filenames or directories.
+	Match []string `json:"match"`
+
+	// Pattern to ignore filenames or directories.
+	Ignore []string `json:"ignore"`
+
+	// Filename to look for the root of a package.
+	Package []string `json:"package"`
+
+	// Packages to always exclude.
+	ExcludePackages []string `json:"exclude-packages"`
+}
+
+func LoadConfig(path string) (Config, error) {
+	bytes, err := os.ReadFile(path)
+	if err != nil {
+		return Config{}, err
+	}
+	config, err := ParseConfig(bytes)
+	if err != nil {
+		return Config{}, err
+	}
+	return config, nil
+}
+
+func ParseConfig(source []byte) (Config, error) {
```

Comment:
issue: provide function-level Godoc comments.

Reply:
question: Is `ParseConfig()` used anywhere other than `LoadConfig()`?

Reply:
Added Godocs.

No, good point. I just made it a local function `loadConfig()`